src/info/guardianproject/otr/app/im/app/ChatFileStore.java

```diff
@@ -162,16 +148,70 @@ public static Bitmap getThumbnailVfs(ContentResolver cr, Uri uri) {
     }
 
     /**
-     * @param mCacheWord
+     * Setup IOCipher VirtualFileSystem without a user-provided password.
+     * @param context
+     */
+    public static void initWithoutPassword(Activity activity) {
+        init(activity, null);
+    }
+
+    /**
+     * Careful! All of the {@code File}s in this method are {@link java.io.File}
+     * not {@link info.guardianproject.iocipher.File}s
+     *
+     * @param context
+     * @param key
+     * @throws IllegalArgumentException
      */
-    public static void init(Context context, String password)  throws IllegalArgumentException {
+    public static void init(Context context, byte[] key) throws IllegalArgumentException {
+        // there is only one VFS, so if its already mounted, nothing to do
+        VirtualFileSystem vfs = VirtualFileSystem.get();
+        if (vfs.isMounted()) {
+            Log.w(TAG, "VFS " + vfs.getContainerPath() + " is already mounted, skipping init()");
+            return;
+        }
+
+        /* TODO None of these key/keyText transformations are necessary since IOCipher/SQLCipher
+         * will handle long strings and blank strings, but changing it might break existing
+         * DBs.  These transformations where gathered from a couple places to be centralized here.
+         */
+        String keyText;
+        if (key == null)
+            keyText = "";
+        else
+            keyText = new String(Hex.encodeHex(key));
+        if (keyText.length() > 32)
+            keyText = keyText.substring(0, 32);
+
+        SharedPreferences settings = PreferenceManager.getDefaultSharedPreferences(context);
+        if (settings.getBoolean(
+                context.getString(R.string.key_store_media_on_external_storage_pref), false))
```

Comment:
If this is false by default, then the first time this pref is set (i.e. on an upgrade where the 'key_store_media_on_external_storage_pref' didn't exist before) would be false, and switch to the internal dbFilePath, which would then be created and mounted.

I think if the pref doesn't exist, we should check if the external path exists first, and then set to true, right?